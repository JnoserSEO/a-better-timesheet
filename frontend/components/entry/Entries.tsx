import React from 'react'
import {Entry, EntryProps} from './Entry'

export interface EntriesProps {
    entries: EntryProps[]
}

export const Entries: React.FC<EntriesProps> = ({entries})=>{
    return <div>
        {entries.map((e, index) => <Entry key={index} hours={e.hours} name={e.name} /> )}
    </div>
}