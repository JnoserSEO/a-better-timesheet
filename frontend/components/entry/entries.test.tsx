import React from 'react'
import { mount } from 'enzyme'
import { Entries } from './Entries'


test('should render Entries with two timesheet entries', () => {
    const wrapper = mount(<Entries entries={[{hours:2,name:'Comp A'},{hours:1,name:'Comp B'}]} />)
    expect(wrapper.find('.timesheet-entry')).toHaveLength(2)
    expect(wrapper.find).toMatchSnapshot()
})