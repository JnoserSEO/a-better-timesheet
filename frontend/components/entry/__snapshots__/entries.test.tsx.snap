// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Entries with two timesheet entries 1`] = `[Function]`;
